python_ta/debug/accumulation_table.py

```diff
@@ -0,0 +1,155 @@
+"""
+Table data structure that prints a nicely formatted table
+for an accumulator loop
+"""
+from __future__ import annotations
+
+import inspect
+import sys
+
+import astroid
+import tabulate
+
+
+def num_whitespaces(start_of_loop: str) -> int:
+    """Number of spaces at the beginning of the accumulation loop"""
+    blank_chars = 0
+    for char in start_of_loop:
+        if char.isspace():
+            blank_chars += 1
+        else:
+            break
+
+    return blank_chars
+
+
+def get_loop_lines(lines: list[str], num_whitespace: int) -> str:
+    """Function that returns the lines from the start to the end
+    of the accumulator loop
+    """
+    endpoint = len(lines)
+    for i in range(len(lines)):
+        if lines[i].strip() != "" and not lines[i][num_whitespace].isspace():
+            endpoint = i
+            break
+
+    return "\n".join(lines[:endpoint])
+
+
+def get_for_node(frame: types.FrameType) -> types.NodeNG:
```

Comment:
`types` is an actual Python module (see its [documentation](https://docs.python.org/3/library/types.html)). You need to import it in order for type checkers (including PyCharm) to verify the actual type.

For the return type, the correct type is `For`, not `NodeNG`. This is also imported from `astroid`, not `types`.